### PR TITLE
EXL-318 Fixed buttons and functionality on Printers view

### PR DESCRIPTION
**Pull request by zsso**

### CODE REVIEW

File: plugin/inc/specialviews/printers.class.php [hunk 1]
```diff
@@ -52,27 +40,27 @@ public static function getTicketStatusDisplay($row_data): string
...
             'list_ticket' => [
-                'link' => "view.php?view=operations&operations0[printer_id]=$row_data[printer_id]&operations0[filter_description]=$operations_filter_description",
+                'link' => "views.php?view=GlpiPlugin\Iservice\Specialviews\Operations&operations0[printer_id]=$row_data[printer_id]&operations0[filter_description]=$operations_filter_description",
                 'icon' => $CFG_GLPI['root_doc'] . '/plugins/iservice/pics/app_detail.png',
                 'title' => __('Operations list', 'iservice'),
                 'visible' => Session::haveRight('plugin_iservice_view_operations', READ),
             ],
             'counters' => [
-                'link' => "view.php?view=printercounters2&printercounters20[supplier_name]=" . urlencode($row_data['supplier_name']),
+                'link' => '/not_imported', //"view.php?view=printercounters2&printercounters20[supplier_name]=" . urlencode($row_data['supplier_name']),
```
Comment (hupupu) [inline plugin/inc/specialviews/printers.class.php:49]:
```suggestion
                'link' => '#', //"view.php?view=printercounters2&printercounters20[supplier_name]=" . urlencode($row_data['supplier_name']),
```

File: plugin/inc/specialviews/printers.class.php [hunk 2]
```diff
@@ -324,8 +312,8 @@ protected function getSettings(): array
     {
         global $CFG_GLPI;
 
-        $printer_counters_button = IserviceToolBox::inProfileArray('client') ? '' :
-            "<a class='vsubmit' href='view.php?view=printercounters2' target='_blank'>" . __('Printer counters', 'iservice') . " v2</a>";
+        $printer_counters_button = IserviceToolBox::inProfileArray('client') ? '' : '';
```
Comment (hupupu) [inline plugin/inc/specialviews/printers.class.php:315]:
```suggestion
        $printer_counters_button = IserviceToolBox::inProfileArray('client') ? '' : '#';
```

File: plugin/inc/views/view.class.php [hunk 3]
```diff
@@ -203,7 +203,7 @@ public function getMachineName(): string
 
     public function getRequestArrayName($add_detail_level = 0): string
     {
-        return $this->getMachineName() . ($this->detail_displaying + $add_detail_level);
+        return strtolower($this->getMachineName()) . ($this->detail_displaying + $add_detail_level);
```
Comment (hupupu) [inline plugin/inc/views/view.class.php:206]: Did you check everywhere in the code where this function is called if this change won't break anything.
How did this work until now without the `strtolower`?

**Review by hupupu — APPROVED** [review]

File: plugin/inc/views/view.class.php (hunk 3 above)
Comment (zsso) [reply plugin/inc/views/view.class.php:206]: Until now the getMachineName() method was different.

old: `function getMachineName() {
        $class_name = get_class($this);
        return strtolower(strpos($class_name, "PluginIserviceView_") === 0 ? substr($class_name, strlen("PluginIserviceView_")) : $class_name);
    }`

new: `public function getMachineName(): string
    {
        return basename(str_replace('\\', '/', get_class($this)));
    }`

**Pull request merged**